tests/acceptance/test_examples.py

```diff
@@ -0,0 +1,60 @@
+"""Acceptance tests for shipping examples."""
+
+import re
+import runpy
+from pathlib import Path
+
+import pytest
+
+import nidaqmx
+
+EXAMPLES_DIRECTORY = Path(__file__).parent.parent.parent / "nidaqmx_examples"
+EXAMPLE_PATHS = [p for p in EXAMPLES_DIRECTORY.glob("**/*.py") if p.name != "__init__.py"]
+
+
+@pytest.mark.parametrize("example_path", EXAMPLE_PATHS)
+def test__shipping_example__run__no_errors(example_path: Path):
+    """Test that a shipping example does not error."""
+    local_system = nidaqmx.system.System.local()
+    example_source = example_path.read_text()
+    for device_name in _find_device_names(example_source):
+        if device_name not in local_system.devices:
+            pytest.skip(f"Cannot find device {device_name}.")
+        device = local_system.devices[device_name]
+        for physical_channel_name in _find_physical_channel_names(example_source, device_name):
+            if not _has_physical_channel(device, physical_channel_name):
+                pytest.skip(
+                    f"Device {device_name} does not have physical channel {physical_channel_name}"
+                )
+    if example_path.name == "ci_pulse_freq.py":
+        pytest.skip("Example times out if there is no signal.")
+    if example_path.name == "every_n_samples_event.py":
+        pytest.skip("Example waits for keyboard input.")
+
+    runpy.run_path(example_path)
```

Comment:
`pytest-cov` fails, probably because this is not a `str`.

```
tests\acceptance\test_examples.py:34:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Users\Administrator\.pyenv\pyenv-win\versions\3.9.13\lib\runpy.py:288: in run_path
    return _run_module_code(code, init_globals, run_name,
C:\Users\Administrator\.pyenv\pyenv-win\versions\3.9.13\lib\runpy.py:97: in _run_module_code
    _run_code(code, mod_globals, init_globals,
C:\Users\Administrator\.pyenv\pyenv-win\versions\3.9.13\lib\runpy.py:87: in _run_code
    exec(code, run_globals)
.venv\lib\site-packages\coverage\control.py:382: in _should_trace
    disp = self._inorout.should_trace(filename, frame)
.venv\lib\site-packages\coverage\inorout.py:328: in should_trace
    filename = source_for_file(dunder_file)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

filename = WindowsPath('X:/nidaqmx-python/nidaqmx_examples/system_properties.py')

    def source_for_file(filename: str) -> str:
        """Return the source filename for `filename`.

        Given a file name being traced, return the best guess as to the source
        file to attribute it to.

        """
>       if filename.endswith(".py"):
E       AttributeError: 'WindowsPath' object has no attribute 'endswith'

.venv\lib\site-packages\coverage\python.py:106: AttributeError
```